src/org/javarosa/core/model/FormDef.java

```diff
@@ -1029,15 +1031,18 @@ public void populateDynamicChoices(ItemsetBinding itemset, TreeReference curQRef
                 copyNode = this.getMainInstance().resolveReference(itemset.copyRef.contextualize(item));
             }
             if (itemset.valueRef != null) {
-                // value = itemset.valueExpr.evalReadable(this.getMainInstance(),
-                // new EvaluationContext(exprEvalContext, item));
                 value = itemset.valueExpr
-                        .evalReadable(fi, new EvaluationContext(exprEvalContext, item));
+                        .evalReadable(formInstance, new EvaluationContext(exprEvalContext, item));
+            }
+
+            // Provide a default value if none is specified
+            value = value != null ? value : "dynamic:" + i;
```

Comment:
I'm curious: can this really happen? If I understand this correctly, a form could define choices without a value?